lib/Subscriber/Phpcr/RemoveSubscriber.php

```diff
@@ -55,12 +56,20 @@ public static function getSubscribedEvents()
      * remove any references to the node.
      *
      * @param RemoveEvent $event
+     *
+     * @throws DocumentReferencedException
      */
     public function handleRemove(RemoveEvent $event)
     {
         $document = $event->getDocument();
         $node = $this->documentRegistry->getNodeForDocument($document);
 
+        $references = iterator_to_array($node->getReferences());
```

Comment:
This should not be done here. It adds a potentially very significant amount of overhead. The implementation, i.e. Jackalope, should handle this, not us.

This makes doing things in batches potentially slow. This should be handled in Sulu AFTER the PHPCR implementation has thrown a Referential integrity exception. Failing that, it should be done in the consuming code (Sulu).